episodes/itksnap.Rmd

```diff
@@ -0,0 +1,630 @@
+#sizeof_hdr      : 348
+#data_type       : b''
+#db_name         : b''
+#extents         : 0
+#session_error   : 0
+#regular         : b''
+#dim_info        : 0
+#dim             : [  3 512 512 175   1   1   1   1]
+#intent_p1       : 0.0
+#intent_p2       : 0.0
+#intent_p3       : 0.0
+#intent_code     : none
+#datatype        : int16
+#bitpix          : 16
+#slice_start     : 0
+#pixdim          : [-1.         0.9765625  0.9765625  2.         1.         1.
+#  1.         1.       ]
+#vox_offset      : 0.0
+#scl_slope       : nan
+#scl_inter       : nan
+#slice_end       : 0
+#slice_code      : unknown
+#xyzt_units      : 2
+#cal_max         : 0.0
+#cal_min         : 0.0
+#slice_duration  : 0.0
+#toffset         : 0.0
+#glmax           : 0
+#glmin           : 0
+#descrip         : b''
+#aux_file        : b''
+#qform_code      : unknown
+#sform_code      : aligned
+#quatern_b       : 0.0
+#quatern_c       : 1.0
+#quatern_d       : 0.0
+#qoffset_x       : 249.51172
+#qoffset_y       : -33.01172
+#qoffset_z       : -553.5
+#srow_x          : [ -0.9765625   0.          0.        249.51172  ]
+#srow_y          : [ -0.          0.9765625   0.        -33.01172  ]
+#srow_z          : [   0.    -0.     2.  -553.5]
+#intent_name     : b''
+#magic           : b'n+1'
+```
+
+### qform   
+```
+print(nii3ct.get_qform())
+## dicom2nifti.convert_directory(dicoms, ".", compression=True, reorient=True) # nii.gz
+#[[  -0.9765625     0.            0.          249.51171875]
+# [   0.            0.9765625     0.          -33.01171875]
+# [   0.            0.            2.         -553.5       ]
+# [   0.            0.            0.            1.        ]]
+## dicom2nifti.convert_directory(dicoms, ".", compression=True, reorient=False) # nii.gz
+#[[  -0.9765625     0.            0.          249.51171875]
+# [   0.           -0.9765625     0.          466.01171875]
+# [   0.            0.            2.         -553.5       ]
+# [   0.            0.            0.            1.        ]]
+
+```
+
+### coordinate system  
+The original DICOM images assume the world coordinate system is LPS (i.e. values increase when moving to the left, posterior, and superior).
+See [here](https://nipy.org/nibabel/dicom/dicom_orientation.html), if you obtained a similar affine transform from the DICOM headers.
+The values in the first two rows corresponding to the x and y dimensions would be the negative of those in the nifti header.
+
+
+### Obtaining image pixel data
+```
+pixel_data=nii3ct.get_fdata()
+
+print(pixel_data.dtype, pixel_data.shape)
+#float64 (512, 512, 175)
+
+plt.imshow(pixel_data[:,:,140], cmap="gray"); plt.show()
+#launch plot window
+```
+
+
+
+## Modifying `NifTi` images and headers with Nibabel
+
+### Using float64 data type
+* The image data for a Nifti1Image object can be accessed using the `get_fdata` function.
+Loading the data from disk and cast it to float64 type before returning the data as a numpy array.
```

Comment:
Are you saying to cast it to float64 or the get_fdata does that automatically?

Reply:
Casting is explained here `* Use the `set_data_dtype` function to set the data type to float32. ` which is differnt that `get_fdata`, hope that make sense.